### ISSUE TITLE
Set active channel update

### ISSUE THREAD

**Issue by seantking** [cosmos/ibc-go#450]
## Summary
On the controller side, we should [set the active channel](https://github.com/cosmos/ibc-go/blob/interchain-accounts/modules/apps/27-interchain-accounts/keeper/keeper.go#L146) on the `OnChanOpenAck` step. On the host side, we should set the active channel on the `OnChanOpenConfirm` step. 

Additionally, `OnChanCloseConfirm` should delete the current active channel mapping.

- [x] Set active channel correctly
- [x] Delete the active channel on channel close 

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by damiannolan** [comment]
> Additionally, OnChanCloseConfirm should delete the current active channel mapping.

This will only delete the active channel mapping for one party. I think we'll have to also delete the active channel mapping in `OnTimeoutPacket`. So the scenario there would be:

1. Controller tries `TrySendTx` 
2. Packet times out.
3. Relayer calls [Timeout](https://github.com/cosmos/ibc-go/blob/main/modules/core/keeper/msg_server.go#L552) gRPC handler on Controller, which will in turn closes the channel because it's `ORDERED`.
4. Relayer calls [ChannelCloseConfirm](https://github.com/cosmos/ibc-go/blob/main/modules/core/keeper/msg_server.go#L439) grpc handler on Host

**Comment by seantking** [comment]
Good catch. Agree, we need to add this! I'm wondering if there are any race conditions we need to consider here, especially now that we are allowing for timeouts again.

**Comment by damiannolan** [comment]
I'll include this in #463 🤝

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
